### PR TITLE
Pagination

### LINKED ISSUE #173: [FEATURE] Adding pages in campgrounds
**Describe the solution you'd like**
Hello
I would like to add paging support to the campgrounds page
Currently all campgrounds are shown in a list we can show 10 at a time and remaining in next pages numbered 1,2 and so on
Pls assign this to me

### PR DESCRIPTION
**Pull request by raj-adi00**
## 📋 Description

This PR adds pagination to sort, search and find nearest page in campgrounds. I am displaying total page. And you can go to any page by changing input. If you input exceeds page limit it won't change and it can't below 1. So you can go directly on any page. There is prev and next button to increase pagination by 1 or decrease by 1.
There is one variable in campgrounds.js at controller which is controlling number of items per page. You just need to update that to change number of contents. In my demo video i set it to 3 but in code i am going to commit i have set it to 10. You can change that varialble named campgroundperPage to control number of items in a page.
Also i have disabled map i didn't have map key so it is not visible in my demo video. I have changed the code back so that won't be issue

## 🔨 Changes Made
I have updated backend code to send data as per page and i have added pagination button. In pagination.js i am controlling all the function.

- fixes #173 
- Earlier there was isssue that if i choose any sort like printAsc (low to high cost) page reloaded and though results were sorted but it was not showing order. So i have made that if user selects any order then also his selected option or search keyword is visible to him.
- I am changing the query to get different page results

## ✅ Checklist

Before submitting the PR, please make sure you have completed the following:

- [x ] My code follows the style guidelines of this project.
- [ x] I have performed a self-review of my own code.
- [x ] I have commented my code, particularly in hard-to-understand areas.
- [x ] I have made corresponding changes to the documentation (if applicable).
- [x ] My changes generate no new warnings.
- [ x] I have tested it locally and it works fine.
- [x ] Any dependent changes have been merged and published in downstream modules.

## 🏷️ Types of Changes

What type of changes does your code introduce? (Check all that apply)

- [x ] Bug fix (non-breaking change which fixes an issue) 🐛
- [x ] New feature (non-breaking change which adds functionality) ✨
- [ ] UI enhancement (non-breaking change which enhances UI) 🎨
- [ ] Documentation update 📚

## 🤝 Related Issues

- Fixes #

## 📸 Screenshots (if applicable)

---

## Reminder:
Please ensure your pull request is made to the `test` branch, not `main`.

*Thank you for contributing to VigneshDevHub! We look forward to reviewing your PR and getting it merged!*

### CODE REVIEW

**Comment by raj-adi00** [comment]
@Vignesh025 video for that

https://github.com/user-attachments/assets/3586004e-e092-41a3-b53d-824842cba1cf

If you want any change please tell me

**Comment by Vignesh025** [comment]
@raj-adi00 The pagination is working well, but the cluster map on the campgrounds page isn’t visible.

**Comment by raj-adi00** [comment]
I don't have map key so it is not visible because I commented that part which was asking for key. But while pushing i un commented it

File: views/campgrounds/index.ejs [hunk 1]
```diff
@@ -1,100 +1,131 @@
...
-    if (navigator.geolocation) {
-        navigator.geolocation.getCurrentPosition((position) => {
-            const userLat = position.coords.latitude;
-            const userLon = position.coords.longitude;
-            // Redirect to the /campgrounds/nearest route with lat and lng as query parameters
-            window.location.href = `/campgrounds/nearest?lat=${userLat}&lng=${userLon}`;
+    <script>
+        const mapToken = '<%- process.env.MAPBOX_TOKEN %>';
+        // const campgrounds = <%- JSON.stringify(campgrounds) %>;
+        // console.log(campgrounds);
```
Comment (Sowmiyajayakumar) [inline views/campgrounds/index.ejs:102]: uncomment these lines

**Comment by raj-adi00** [comment]
@Sowmiyajayakumar uncommented those lines

**Comment by Vignesh025** [comment]
@raj-adi00 , Resolve the conflicts.

**Comment by Vignesh025** [comment]
@raj-adi00 The pagination isn’t working now. Could you please check it out?

**Comment by raj-adi00** [comment]
@Vignesh025 check it now.

**Comment by Vignesh025** [comment]
@raj-adi00 The search functionality isn’t working. Please make sure it’s fully functional and working as expected.

**Pull request closed without merging**